src/platform/intel/cavs/lib/memory.c

```diff
@@ -74,40 +76,40 @@ static SHARED_DATA struct block_map sys_rt_heap_map[CONFIG_CORE_COUNT][3] = {
 };
 
 /* Heap blocks for modules */
-static SHARED_DATA struct block_hdr mod_block64[HEAP_RT_COUNT64];
-static SHARED_DATA struct block_hdr mod_block128[HEAP_RT_COUNT128];
-static SHARED_DATA struct block_hdr mod_block256[HEAP_RT_COUNT256];
-static SHARED_DATA struct block_hdr mod_block512[HEAP_RT_COUNT512];
-static SHARED_DATA struct block_hdr mod_block1024[HEAP_RT_COUNT1024];
-static SHARED_DATA struct block_hdr mod_block2048[HEAP_RT_COUNT2048];
-static SHARED_DATA struct block_hdr mod_block4096[HEAP_RT_COUNT4096];
+static SHARED_DATA struct block_hdr mod_block64[HEAP_COUNT64 * RT_TIMES_MAX];
+static SHARED_DATA struct block_hdr mod_block128[HEAP_COUNT128 * RT_TIMES_MAX];
+static SHARED_DATA struct block_hdr mod_block256[HEAP_COUNT256 * RT_TIMES_MAX];
+static SHARED_DATA struct block_hdr mod_block512[HEAP_COUNT512 * RT_TIMES_MAX];
+static SHARED_DATA struct block_hdr mod_block1024[HEAP_COUNT1024 * RT_TIMES_MAX];
+static SHARED_DATA struct block_hdr mod_block2048[HEAP_COUNT2048 * RT_TIMES_MAX];
+static SHARED_DATA struct block_hdr mod_block4096[HEAP_COUNT4096 * RT_TIMES_MAX];
```

Comment:
"times" seems a bit unclear - count?

Reply:
um, we already have HEAP_COUNTxxxx, so.., "times" means multiplying here.

Reply:
yeah, I see it was already used before this patch...